docs/architecture/adr-011-reports-module.md

```diff
@@ -0,0 +1,307 @@
+message MsgCreateReport {
+  // Id of the reason this report has been created for
+  required uint32 reason_id = 1;
+
+  // Message attached to this report
+  optional string message = 2;
+
+  // Address of the reporter 
+  required string reporter = 3;
+
+  // Target of the report
+  oneof Target {
+    UserData user_data = 4;
+    PostData post_data = 5;
+  }
+}
+
+// MsgCreateReportResponse represents the Msg/CreateReport response type
+message MsgCreateReportResponse {
+  // Id of the newly created report
+  required uint64 report_id = 1;
+}
+
+// MsgDeleteReport represents the message to be used when deleting a report
+message MsgDeleteReport {
+  // Id of the subspace that contains the report to be deleted 
+  required uint64 subspace_id = 1;
+  
+  // Id of the report to be deleted 
+  required uint64 report_id = 2; 
+  
+  // Address of the user deleting the report 
+  required string signer = 3;
+}
+
+// MsgDeleteReportResponse represents the Msg/DeleteReport response type
+message MsgDeleteReportResponse {}
+
+// MsgSupportReason represents the message to be used when wanting to support a reason from the module params
+message MsgSupportReason {
+  // Id of the subspace for which to support the reason
+  required uint64 subspace_id = 1;
+  
+  // Id of the reason that should be supported
+  required uint32 reason_id = 2;
+  
+  // Address of the user signing the message
+  required string signer = 3;
+}
+
+// MsgSupportReasonResponse represents the Msg/SupportReason response type
+message MsgSupportReasonResponse {}
+
+// MsgAddReason represents the message to be used when adding a new supported reason
+message MsgAddReason {
+  // Id of the subspace for which to add the reason 
+  required uint64 subspace_id = 1;
+  
+  // Title of the reason
+  required string title = 2;
+
+  // Extended description of the reason and the cases it applies to
+  optional string description = 3;
+  
+  // Address of the user adding the supported reason
+  required string signer = 4;
+}
+
+// MsgAddReasonResponse represents the Msg/AddReason response type
+message MsgAddReasonResponse {
+  // Id of the newly supported reason
+  required uint32 reason_id = 1;
+}
+
+// MsgRemoveReason represents the message to be used when removing an exiting reporting reason
+message MsgRemoveReason {
+  // Id of the subspace from which to remove the reason 
+  required uint64 subspace_id = 1;
+
+  // Id of the reason to be deleted
+  required uint32 reason_id = 2;
+
+  // Address of the user adding the supported reason
+  required string signer = 3;
+}
+
+// MsgRemoveReasonResponse represents the Msg/RemoveReason response type
+message MsgRemoveReasonResponse {
+  // Id of the newly supported reason
+  required uint32 reason_id = 1;
+}
+
+```
+
+### `Query` Service
+```protobuf
+service Query {
+  // Reports allows to query the reports for a specific target
+  rpc Reports(QueryReportsRequest) returns (QueryReportsResponse) {
```

Comment:
It would be useful to have `ReportTallyResult` to show the tally result. Moderator group can use the result to govern their subspaces.